### PR TITLE
remove calls to internal dev debugger in training- and eval loop

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Replaces the calls to the internal dev debugger in the loops. Tests are updated to cover what the "debugger" offered for tracking the calls inside the loops. 
This is just one step towards removing the `InternalDebugger` completely, and to make loops more easily extensible.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: tests/trainer/test_dataloaders.py [hunk 1]
```diff
@@ -551,10 +538,6 @@ def test_dataloaders_with_fast_dev_run(tmpdir, fast_dev_run):
         trainer.test(model)
         assert trainer.num_test_batches == [fast_dev_run] * len(trainer.test_dataloaders)
 
-        # verify sanity check batches match as expected
-        num_val_dataloaders = len(model.val_dataloader())
-        assert trainer.dev_debugger.num_seen_sanity_check_batches == trainer.num_sanity_val_steps * num_val_dataloaders
-
```
Comment (awaelchli) [inline tests/trainer/test_dataloaders.py:557]: reviewer: already covered by the test in `tests/trainer/test_trainer.py::test_num_sanity_val_steps`

File: tests/trainer/test_dataloaders.py [hunk 2]
```diff
@@ -446,7 +446,6 @@ def test_dataloaders_with_limit_percent_batches(tmpdir, limit_train_batches, lim
 @pytest.mark.parametrize(
     ["limit_train_batches", "limit_val_batches", "limit_test_batches"], [(0, 0, 0), (1, 2, 3), (1, 2, 1e50)]
 )
-@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
 def test_dataloaders_with_limit_num_batches(tmpdir, limit_train_batches, limit_val_batches, limit_test_batches):
```
Comment (awaelchli) [inline tests/trainer/test_dataloaders.py:449]: reviewer: I'm updating this to make use of unittest facilities instead of our internal debugger

**Review by carmocca — APPROVED** [review]
Awesome!

**Review by tchaton — APPROVED** [review]
Love it ! Let's drop support for the DevDebugger entirely no?

**Comment by awaelchli** [comment]
Yes, that's the plan, incrementally with one PR at a time :)

**Review by justusschock — APPROVED** [review]

**Pull request merged**